### ISSUE TITLE
Wrong order of async plugins execution

### ISSUE THREAD

**Issue by lenarakhmadeev** [postcss/postcss#269]

**Comment by lenarakhmadeev** [comment]
```js
gulp.task('async', function () {
    var processerer = function (message, timeout) {
        return function (css) {
            new Promise(function (resolve) {
                setTimeout(function () {
                    console.log(message);
                    resolve(css);
                }, timeout);
            });
        }
    };

    var processors = [
        processerer('1m', 4000),
        processerer('2m', 2000),
        processerer('3m', 3000)
    ];

    return gulp.src('demo/demo.css')
        .pipe( postcssAsync(processors) );
});
```

```
mbp-lenar:postcss-image-mockup lenar$ gulp async
[12:40:31] Using gulpfile ~/dev/postcss-image-mockup/gulpfile.js
[12:40:31] Starting 'async'...
[12:40:31] Finished 'async' after 20 ms
2m
3m
1m
```

**Comment by MoOx** [comment]
Do you want async to be sync ?!

**Comment by w0rm** [comment]
gulp-postcss doesn't support async plugins

**Comment by ai** [comment]
@MoOx we must run async plugins in sync way, because they can have a conflict while editing CSS AST.

**Comment by ai** [comment]
@w0rm custom Gulp processor was used here.

**Comment by ai** [comment]
@lenarakhmadeev you forgot to return Promise. But seems like like there is other issue in PostCSS

**Comment by ai** [comment]
Fixed 07b3914

**Comment by w0rm** [comment]
@ai, oh, I see! Anyway, I have created a [ticket for this](https://github.com/w0rm/gulp-postcss/issues/20)
